src/views/MainView.js

```diff
@@ -55,10 +68,155 @@
+      var fields = ['Entry Type', 'Label'];
+      for (var index in entries){
+        var entry = entries[index];
+        this.addFieldFromEntry(entry, fields);
+      }
+      return fields;
+    },
+    addDataFromEntry: function(entry, fields){
+      var entry_data = {};
+      entry.items.forEach(function(item) {
+        var field = item.key;
+        entry_data[field] = item.value;
+      });
+      entry_data['Entry Type'] = entry.type;
+      entry_data['Label'] = entry.label;
+      return entry_data;
+    },
+    getCsvData: function(entries, fields) {
+      var data = [];
+      for (var index in entries){
+        var entry = entries[index];
+        data.push(this.addDataFromEntry(entry, fields));
+      }
+      return data;
+    },
+    writeCordovaFile: function(fileName, data) {
+      data = JSON.stringify(data, null, '\t');
+      var promise = $.Deferred();
+      window.resolveLocalFileSystemURL(cordova.file.cacheDirectory, function (directoryEntry) {
+          directoryEntry.getFile(fileName, { create: true }, function (fileEntry) {
+              fileEntry.createWriter(function (fileWriter) {
+                  var blob = new Blob([data], { type: 'text/csv' });
+                  fileWriter.write(blob);
+                  promise.resolve(fileEntry.fullPath);
+              }, console.log);
+          }, console.log);
+      }, console.log);
+      return promise;
+    },
+    saveCsv: function(csv){
+      var type = 'text/csv';
+      var file = new Blob([csv], {type: type});
+      var a = document.createElement("a");
+      var url = window.URL.createObjectURL(file);
+      a.href = url;
+      a.target="_blank";
+      a.download = 'export.csv';
+      a.hidden = true;
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+      window.URL.revokeObjectURL(url);
+    },
+    generateCsvFromEntries: function(entries) {
+      var fields = this.getCsvFields(entries);
+      var data = this.getCsvData(entries, fields);
+      var csv = json2csv({'data': data, 'fields': fields});
+      return csv;
+    },
+    getEntry: function (entry) {
+      var entry_model = new window.app.EntryModel(entry);
+      var promise = $.Deferred();
+      entry_model.fetch({success: function(_, resp){
+        promise.resolve(resp);
+      }});
+      return promise;
+    },
+    getEntries: function() {
+      var self = this;
+      var username = window.app.accountModel.get("username");
+      var hashArray = window.sjcl.hash.sha256.hash(username);
+      var hash = window.sjcl.codec.hex.fromBits(hashArray);
+      var encryptedIndexJSON = window.localStorage.getItem("encryptr-" + hash + "-index");
+      if (encryptedIndexJSON && window.app.accountModel.get("passphrase")) {
+        try {
+          var decryptedIndexJson =
+            window.sjcl.decrypt(window.app.accountModel.get("passphrase"),
+                                encryptedIndexJSON, window.crypton.cipherOptions);
```

Comment:
Ok, I would say the code should be refactored rather than copied, would it be simple to create a small utils that handles all of this and get that to be used in both places?